action.yml

```diff
@@ -34,11 +34,26 @@ runs:
         curl -L https://github.com/pantheon-systems/terminus/releases/download/$TERMINUS_RELEASE/terminus.phar --output terminus
         chmod +x terminus
         sudo ln -s ~/terminus/terminus /usr/local/bin/terminus
+        mkdir -p ~/.terminus/{cache,plugins}
       env:
         TERMINUS_RELEASE: ${{ inputs.terminus-version || env.TERMINUS_RELEASE }}
 
+    - uses: actions/cache@v3
+      id: terminus-cache
+      ## description: "Cache Terminus cache directory to eliminate auth API limits."
+      with:
+        path: $HOME/.terminus/
+        key: ${{ runner.os }}-terminus
+        restore-keys: |
+            ${{ runner.os }}-terminus-
+
     - name: Login to Pantheon
       if: ${{ inputs.pantheon-machine-token }}
       shell: bash
+      ## description: "Authenticate with Pantheon using a machine token only if a session has not already been established"
       run: |
-        terminus auth:login --machine-token="${{ inputs.pantheon-machine-token }}"
+        if [ -f ~/.terminus/cache/session ]; then
+          terminus auth:login
+        else
+          terminus auth:login --machine-token="${{ inputs.pantheon-machine-token }}"
+        fi
```

Comment:
I tried to make this suggestion in Slack, but folks seemed to think I didn't understand the caching component. 😄 

I think that if you rewrote lines 55-59 with `terminus auth:login || terminus auth:login --machine-token="${{ inputs.pantheon-machine-token }}"`, then I think it would work if the cached session does not exist, or if the cached session exists but has expired, or if the cached session exists and is still valid.

The use case for multiple machine tokens is: you revoke the machine token you are using in CI because it was compromised, and you add a new (single) machine token. I think this implementation fails in that scenario, because it will keep using the old session. Or maybe the old session still works for a while, but eventually the cached session will expire, and then this algorithm doesn't provide a good way to refresh the session using the machine token.

Reply:
Note that I have not tried my suggestion; it only works if `terminus auth:login` returns a status code 1 when the cached session does not exist or has expired. Hopefully it does that.

Reply:
I like all of this - I can't believe that use-case for multiple machine tokens didn't occur to me, because we have a slightly alternative version: since right now machine tokens are per-user rather than per-site, and there's a high overhead in maintaining a dedicated "machine user" for each site, we've got a single token as an organisation secret so that we can rotate it as often as we want (yearly, monthly, weekly or even daily)